org.libreoffice.LibreOffice.BundledExtension.Voikko

stbergmann opened the pull request:
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

nedrichards commented:
bot, build org.libreoffice.LibreOffice.BundledExtension.Voikko

hfiguiere commented on org.libreoffice.LibreOffice.BundledExtension.Voikko.json:
```diff
@@ -0,0 +1,80 @@
+    "sdk": "org.freedesktop.Sdk//20.08",
+    "build-extension": true,
+    "separate-locales": false,
+    "appstream-compose": false,
```
```suggestion
    "appstream-compose": false,
    "build-options": {
        "prefix": "/app/libreoffice/share/extensions/Voikko",
        "prepend-path": "/app/libreoffice/share/extensions/Voikko/bin"
    },
```

hfiguiere commented on org.libreoffice.LibreOffice.BundledExtension.Voikko.json:
```diff
@@ -0,0 +1,80 @@
+    "modules": [
+        {
+            "name": "libvoikko",
+            "buildsystem": "simple",
+            "build-commands": [ "./configure --prefix=/app/libreoffice/share/extensions/Voikko/build-deps/libvoikko --disable-hfst --with-dictionary-path=/app/libreoffice/share/extensions/Voikko/voikko",
+                                "make",
+                                "make install",
+                                "mkdir -p /app/libreoffice/share/extensions/Voikko/voikko/Linux",
+                                "cp -L /app/libreoffice/share/extensions/Voikko/build-deps/libvoikko/lib/libvoikko.so.1 /app/libreoffice/share/extensions/Voikko/voikko/Linux/",
+                                "mkdir -p /app/libreoffice/share/extensions/Voikko/pythonpath",
+                                "cp python/libvoikko.py /app/libreoffice/share/extensions/Voikko/pythonpath/" ],
```
```suggestion
            "buildsystem": "autoconf",
            "config-opts": [
                "--disable-hfst",
                "--with-dictionary-path=${FLATPAK_DEST}/voikko"
            ],
            "post-install": [ 
                "install -d ${FLATPAK_DEST}/pythonpath",
                "cp python/libvoikko.py ${FLATPAK_DEST}/pythonpath/" 
            ],
            "cleanup": [
                "/bin"
            ],
```

hfiguiere commented on org.libreoffice.LibreOffice.BundledExtension.Voikko.json:
```diff
@@ -0,0 +1,80 @@
+        {
+            "name": "voikko-fi",
+            "buildsystem": "simple",
+            "build-commands": [ "PATH=$PATH:/app/libreoffice/share/extensions/Voikko/build-deps/libvoikko/bin:/app/libreoffice/share/extensions/Voikko/build-deps/foma/bin make vvfst", "make vvfst-install DESTDIR=/app/libreoffice/share/extensions/Voikko/voikko" ],
```
```suggestion
            "build-commands": [ 
                "make vvfst", 
                "make vvfst-install" 
            ],
```

hfiguiere commented on org.libreoffice.LibreOffice.BundledExtension.Voikko.json:
```diff
@@ -0,0 +1,80 @@
+            "name": "libreoffice-voikko",
+            "buildsystem": "simple",
+            "build-commands": [ "make oxt",
+                                "mkdir -p /app/libreoffice/share/extensions/Voikko",
+                                "unzip -d /app/libreoffice/share/extensions/Voikko build/voikko.oxt" ],
```
```suggestion
                                "unzip -d ${FLATPAK_DEST} build/voikko.oxt" ],
```

hfiguiere commented on org.libreoffice.LibreOffice.BundledExtension.Voikko.json:
```diff
@@ -0,0 +1,80 @@
+            ]
+        }
+    ],
+    "cleanup-commands": [ "rm -r /app/libreoffice/share/extensions/Voikko/build-deps" ]
```
```suggestion
```

stbergmann commented on flathub.json:
```diff
@@ -0,0 +1,3 @@
+{
+    "skip-arches": ["arm"]
```
Where do you see a flathub.json file in <https://github.com/flathub/flathub/commit/e2aaf7193988e19304d5e036ef5db57f924c99e0> "Provide Voikko as a bundled extension for LibreOffice"?  (Note that I had erroneously pushed that first as <https://github.com/flathub/flathub/commit/d1c7539fa9e340a59d5d435310116a90b4ba9878> "Provide Voikko as a bundled extension for LibreOffice" based on stale content from an old stbergmann/flathub branch, and then force-pushed the correction.)

stbergmann commented:
@hfiguiere, thanks for the suggestions.  I liked the original way of clearly separating the build-time dependencies under ${FLATPAK_DEST}/build-deps/ instead of mangling their artefacts into ${FLATPAK_DEST} itself (and whose final content has a radically different directory structure from the traditional Unix bin, include, lib, share structure).  But sure, it simplifies some of the build script lines, and if you think that's better, so be it.  Incorporated your suggestions, with necessary fixes, in <https://github.com/flathub/flathub/commit/7784e57d8edcb2b81c462d234b65b52e4b19b48d> "Provide Voikko as a bundled extension for LibreOffice" (and added the "The upstream pythonpath/VoikkoHandlePool.py..." paragraph to the commit message, which I had originally forgotten).

hfiguiere commented:
bot, build org.libreoffice.LibreOffice.BundledExtension.Voikko

stbergmann commented:
bot, build org.libreoffice.LibreOffice.BundledExtension.Voikko

stbergmann commented:
bot, build org.libreoffice.LibreOffice.BundledExtension.Voikko

hfiguiere commented:
You need add a `flathub.json` like there: https://github.com/flathub/org.gimp.GIMP.Plugin.GMic/blob/master/flathub.json to tell Flathub to legitimately ignore the lack of icon

stbergmann commented:
bot, build org.libreoffice.LibreOffice.BundledExtension.Voikko

nedrichards commented:
/merge

jpakkane commented:
Should this work in the released version? There have been several updates to LO since this was merged but when I open a Finnish document in LO it always says "missing hyphenation data". The extension manager does not show the Voikko extension either.,

stbergmann commented:
The issue that the Voikko extension does not show up on the page of LibreOffice from Flathub in e.g. GNOME Software is being discussed on the Matrix #flatpak channel.  It appears that for now you have to install it manually.

hfiguiere commented:
extensions are not installed automatically as per the libreoffice manifest :

`            "no-autodownload": true,`

And for GNOME software, it's management of extensions is greatly lacking. I have a few issues open.

jpakkane commented:
Installing it manually means building from source as upstream does not provide a prebuilt extension for Linux but instead say that "on Linux the extension is provided by the distribution" as explained [on this page](https://voikko.puimula.org/linux-jakelut.html) (though only in Finnish).

hfiguiere commented:
Now it's visible in GNOME Software:

![image](https://user-images.githubusercontent.com/114441/131422600-15f92abe-bece-413a-b15e-a3222bcbb487.png)

So, problem solved.

hfiguiere commented:
you still need in install it "manually", ie to opt to install it.

hfiguiere commented:
Just to clarify, it was fixed by https://github.com/flathub/org.libreoffice.LibreOffice.BundledExtension.Voikko/pull/1

There was something broken.

stbergmann commented:
@hfiguiere Thanks for spotting and fixing that silly typo!
@jpakkane Just to clarify that with "install it manually" I meant some `flatpak install org.libreoffice.LibreOffice.BundledExtension.Voikko` on the command line; sorry for being too terse there.

hfiguiere commented:
Or by clicking in GNOME Software as shown above.

jpakkane commented:
Can confirm that this works now. Thanks.
